runconfig/parse_unix.go

```diff
@@ -7,51 +7,53 @@ import (
 	"strings"
 )
 
-func parseNetMode(netMode string) (NetworkMode, error) {
-	parts := strings.Split(netMode, ":")
+// ValidateNetMode ensures that the various combinations of requested
+// network settings are valid.
+func ValidateNetMode(c *Config, hc *HostConfig) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
+		return nil
+	}
```

Comment:
@mavenugo I'm not sure I agree with that statement. As it actually happens, from what I have seen, very few commands do NOT pass in a hostConfig. As it is, this is backwards compatible :) "Default" recall is relatively new, so that would actually break backwards compatibility.